src/ray/object_manager/plasma/store.cc

```diff
@@ -972,6 +958,9 @@ void PlasmaStore::SubscribeToUpdates(const std::shared_ptr<Client> &client) {
 Status PlasmaStore::ProcessMessage(const std::shared_ptr<Client> &client,
                                    fb::MessageType type,
                                    const std::vector<uint8_t> &message) {
+  // Global lock is used here so that we allow raylet to access some of methods
+  // that are required for object spilling directly without releasing a lock.
+  std::lock_guard<std::recursive_mutex> guard(mutex_);
```

Comment:
Can we add a test that no we don't attempt to spill these kind of objects? (add API to check bytes spilled and call from python)?

Reply:
I am adding a test without having this stats API. I am checking if the spill temp is empty. I prefer to implement stats API in the next PR (which I can make it by tomorrow or Wed), but if you'd like it to be included in this PR, I can do that.